src/Services/DocumentSign/SignRequest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+
+namespace Tychovbh\Mvc\Services\DocumentSign;
+
+use GuzzleHttp\Client;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Arr;
+
+class SignRequest implements DocumentSignInterface
+{
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    public function create(UploadedFile $file, string $webhook = null)
+    {
+        try {
+            $contents = $file->get();
+
+            $response = $this->request('post', '/documents', [
+                'file_from_content' => base64_encode($contents),
+                'file_from_content_name' => $file->getClientOriginalName(),
+                'events_callback_url' => $webhook
+            ]);
+
+            return [
+                'id' => Arr::get($response, 'uuid', ''),
+                'status' => Arr::get($response, 'status', '')
+            ];
+        } catch (\Exception $exception) {
+            $message = $exception->getMessage();
+        }
+    }
+
+    public function sign(string $id, array $sender, array $signers, string $message = '')
```

Comment:
it is not clear to me how to add a sender and how to add signers, we are improving this:
- replace sender for $from_name and $from_email
- remove $signers from params;

- Add a class property private $signers = []
- Add a class method called signer()
- method params: required email, optional: firstname, lastname
- every time you call this method you will add a record to the $signers property
- add the end return $this; so you can keep calling methods in this class.
- make a check here in de the sign method if this->signers is empty throw error with message "At  least one signerr is required (see addSigner method)"

Your goal is the following:
```php
$signRequest = new SignRequest();
$signRequest->signer('test1@live.com', 'test', '1')
->signer('test2@live.com', 'test', '2')
->sign('3343', 'sender@live.com', 'sender')
```